libyul/Object.cpp

```diff
@@ -61,13 +61,48 @@ string Object::toString(Dialect const* _dialect) const
 	return "object \"" + name.str() + "\" {\n" + indent(inner) + "\n}";
 }
 
-set<YulString> Object::dataNames() const
+set<YulString> Object::qualifiedDataNames() const
 {
-	set<YulString> names;
-	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
+	set<YulString> qualifiedNames = {name};
+	for (shared_ptr<ObjectNode> const& subObjectNode: subObjects)
+	{
+		qualifiedNames.insert(subObjectNode->name);
+		if (auto const* subObject = dynamic_cast<Object const*>(subObjectNode.get()))
+			for (YulString const& subSubObj: subObject->qualifiedDataNames())
+				qualifiedNames.insert(YulString{subObject->name.str() + "." + subSubObj.str()});
+	}
+
 	// The empty name is not valid
-	names.erase(YulString{});
-	return names;
+	qualifiedNames.erase(YulString{});
+
+	return qualifiedNames;
+}
+
+tuple<bool, vector<size_t>> Object::pathToSubObject(string _qualifiedName) const
+{
+	if (_qualifiedName == name.str())
+		return {};
```

Comment:
```suggestion
		return {false, {}};
```
I think this form would make it easier to see at a glance where all the places that can return with `false` are.